webpack/assets/javascripts/foreman_tools.js

```diff
@@ -1,12 +1,15 @@
 import $ from 'jquery';
 import URI from 'urijs';
 
+import store from './react_app/redux';
+import * as LayoutActions from './react_app/components/Layout/LayoutActions';
+
 export function showSpinner() {
-  $('#turbolinks-progress').show();
+  store.dispatch(LayoutActions.showLoading());
```

Comment:
I wonder, why dispatch via the store here vs just import the store on the actions ? (similar to the https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/foreman_toast_notifications.js#L12)

Reply:
just to follow up, what concerned me was the fact that store is now exposed to everything under foreman_tools, I suggest to:
1. move the show/hide spinner functions to foreman_navigations
2. add deprecation warnings to show/hide if they are still being called via foreman_tools
3. expose the store only in foreman_navigations (and you can probably remove the existing content of that file as its all relate to the jquery implementation of the vertical navigation).